### PR TITLE
[RFC] man.vim: remove terminal escape characters

### PR DESCRIPTION
**Pull request by nhooyr**
This will make man.vim work on nixOS.

\cc @teto

### CODE REVIEW

File: runtime/autoload/man.vim [hunk 1]
```diff
@@ -125,8 +125,8 @@ function! s:put_page(page) abort
   setlocal noreadonly
   silent keepjumps %delete _
   silent put =a:page
-  " Remove all backspaced characters.
-  execute 'silent keeppatterns keepjumps %substitute,.\b,,e'.(&gdefault?'':'g')
+  " Remove all backspaced/escape characters.
+  execute 'silent keeppatterns keepjumps %substitute,.\b\|\e\[\d\+m,,e'.(&gdefault?'':'g')
```
Comment (justinmk) [inline runtime/autoload/man.vim]: Might want to group this to be sure the `|` applies correctly.

    \b\|\(\e\[\d\+m\)

Comment (nhooyr) [inline runtime/autoload/man.vim]: I tested it to make sure, I can add it though if you think it's still a good idea.

Comment (justinmk) [inline runtime/autoload/man.vim]: Without the grouping it could match `\b[\d\+m`, if I'm not mistaken.

Comment (nhooyr) [inline runtime/autoload/man.vim]: I tested it again and it works fine. Also, in `:h /bar` it says `A pattern is one or more branches, separated by "\|".` so it shouldn't match `\b[\d\+m`.

Comment (jamessan) [inline runtime/autoload/man.vim]: Yeah, the current pattern will work fine.

**Pull request merged**